### PR TITLE
[SYCL-MLIR][NFC]: Fix compiler warnings

### PR DESCRIPTION
**Pull request by etiotto**
Fix g++ 11.2.1 20220127 (Red Hat 11.2.1-9) compilation warnings.

### CODE REVIEW

File: polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp [hunk 1]
```diff
@@ -1310,9 +1328,8 @@ std::optional<MemoryAccessMatrix> MemoryAccessAnalysis::buildAccessMatrix(
   // 'expr' corresponding to the variable 'factor'.
   auto createMatrixEntry = [&](Value expr, Value factor, size_t row,
                                size_t col) -> bool {
-    assert(row >= 0 && row < accessMatrix.getNumRows() && "row out of bound");
-    assert(col >= 0 && col < accessMatrix.getNumColumns() &&
-           "col out of bound");
+    assert(row < accessMatrix.getNumRows() && "row out of bound");
```
Comment (etiotto) [inline polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp:1331]: size_t is unsigned, row >= 0 always evaluate to true.

**Review by whitneywhtsang — APPROVED** [review]

**Review by sommerlukas — APPROVED** [review]

File: polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp [hunk 2]
```diff
@@ -535,8 +535,11 @@ getOffset(const Value expr, const SmallVectorImpl<Value> &loopAndThreadVars,
 // MemoryAccessMatrix
 //===----------------------------------------------------------------------===//
 
-[[maybe_unused]] raw_ostream &
-mlir::polygeist::operator<<(raw_ostream &os, const MemoryAccessMatrix &matrix) {
+namespace mlir {
+namespace polygeist {
```
Comment (victor-eds) [inline polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp:539]: Using namespace qualifiers is what the [LLVM coding standards](https://llvm.org/docs/CodingStandards.html#use-namespace-qualifiers-to-implement-previously-declared-functions) say, so I think we should stick to that

File: polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp [hunk 3]
```diff
@@ -971,14 +977,20 @@ bool MemoryAccessMatrix::hasReverseStridedOverlappedAccessPattern() const {
 // OffsetVector
 //===----------------------------------------------------------------------===//
 
-[[maybe_unused]] raw_ostream &
-mlir::polygeist::operator<<(raw_ostream &os, const OffsetVector &vector) {
+namespace mlir {
+namespace polygeist {
```
Comment (victor-eds) [inline polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp:981]: Same here

**Review by victor-eds — COMMENTED** [review]
LGTM after suggestions

File: polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp (hunk 2 above)
Comment (etiotto) [reply polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp:539]: Yes I know but that doesn't work and g++ reports a warning. I changed it to get rid of the warning.

File: polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp (hunk 3 above)
Comment (etiotto) [reply polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp:981]: See response at line 539.

**Pull request merged**